### ISSUE TITLE
Collapse Mono builds into single builds for Desktop targets

### ISSUE THREAD

**Issue by safern** [dotnet/runtime#45521]
Currently we fan out test legs and build Mono and Libraries in a separate leg. Then put those together in a run test leg and submit to helix. We do similar things for the mono runtime tests.

However this model has proven that reliability issues like networking hiccups, artifacts service outages, etc, bite us and the gain in time for Mono targets is not worth it. 

We should be collapsing the Mono desktop targets into single legs that build Mono, Libraries, Tests and submit to helix, similar to how we do it for Android, WASM, iOS. That will reduce the number of legs we have on CI and improve reliability for those legs on infra issues.

I will take care of collapsing the libraries tests legs, @naricc could you help collapsing the runtime ones? I guess this would replace: https://github.com/dotnet/runtime/issues/43952

cc: @akoeplinger @ViktorHofer

**Comment by naricc** [comment]
Yes. I will help with the runtime ones. So this basically means that all the mono legs should be using the global-build-job.yml?

Also any preference if I fix this at the same time as https://github.com/dotnet/runtime/issues/43952 or in seperate PRs?

**Comment by safern** [comment]
Yeah, that means they should use the global build template. Just like the Android and Browser runtime tests.

I think it needs to be on a same PR so that we don't have to build coreclr as part of that job.

**Comment by ViktorHofer** [comment]
@naricc were you able to look into this?

**Comment by naricc** [comment]
@ViktorHofer  Sorry, I haven't made any progress on this; its been a lower priority item for me that I keep meaning to get to.

**Comment by SamMonoRT** [comment]
Not going to get to it in 6.0, moving to 7.0.0. We might consider this after RC1 if we have spare cycles.

**Comment by SamMonoRT** [comment]
This is nice to have, but pretty low on priority. Closing this issue.

**Comment by ViktorHofer** [comment]
@SamMonoRT this will become more important with the upcoming VMR efforts. I don't think we should close this issue.

**Comment by SamMonoRT** [comment]
@ViktorHofer @steveisok  Re-opening, but not something anyone on my team maybe familiar with.

**Comment by steveisok** [comment]
This is done on the official build side at least.

**Comment by ViktorHofer** [comment]
And we should probably do this on the PR side as well just for the sake of less legs and reduced resources and time.
